other/install.php

```diff
@@ -1303,15 +1303,46 @@ function DatabasePopulation()
 		{
 			// www isn't really a subdomain in this sense, so strip it out
 			$url_parts['host'] = str_ireplace('www.', '', $url_parts['host']);
-			$pos1 = strrpos($url_parts['host'], '.');
-			if ($pos1 !== false)
+
+			// If host only contains one period there can't be any subdomain
+			if (substr_count($url_parts['host'], '.') > 1)
 			{
-				// 2nd period from the right indicates you have a subdomain
-				$pos2 = strrpos(substr($url_parts['host'], 0, $pos1 - 1), '.');
-				if ($pos2 !== false)
+				$allowed_tld = download_ccSLD_list();
+				if (!empty($allowed_tld))
+				{
+					foreach($allowed_tld as $value)
+					{
+						$pattern = '/\.' . strtr($value, array('.' => '\.', '*' => '[^.]*?')) . '$/';
+						if (preg_match($pattern, $url_parts['host'], $matches, PREG_OFFSET_CAPTURE) === 1)
+						{
+							$stripped_domain = substr($url_parts['host'], 0, $matches[0][1]);
+							$tld = $matches[0][0];
+							break;
+						}
+					}
```

Comment:
Right now, this requires running possibly thousands of `preg_match()` calls in order to test each ccSLD individually. If instead you use `build_regex()` from Subs.php to create a single regex that will match all possible ccSLD values, you will only need to call `preg_match()` once.

Reply:
This should do the job.
```suggestion
if (!empty($allowed_tld))
{
	foreach (build_regex($allowed_tld, '/', true) as $ccSLD_regex)
	{
		preg_match('/\.' . strtr($ccSLD_regex, array('\*' => '[^.]*?')) . '$/u', 'vineyard.ca', $matches, PREG_OFFSET_CAPTURE);

		if (!empty($matches))
			break;
	}
}
```
The ccSLD list is so large that I had to set the `$returnArray` parameter of `build_regex()` to true and the loop through the four regular expressions it returned.

P.S.: Make sure to test this against 5bf8672 or later.

Reply:
This should do the job.
```suggestion
					foreach (build_regex($allowed_tld, '/', true) as $ccSLD_regex)
					{
						preg_match('/\.' . strtr($ccSLD_regex, array('\*' => '[^.]*?')) . '$/u', $url_parts['host'], $matches, PREG_OFFSET_CAPTURE);

						if (!empty($matches))
							break;
					}
					$stripped_domain = substr($url_parts['host'], 0, $matches[0][1]);
					$tld = $matches[0][0];
```
The ccSLD list is so large that I had to set the `$returnArray` parameter of `build_regex()` to true and the loop through the four regular expressions it returned.

P.S.: Make sure to test this against 5bf8672 or later.